Test/ApiTests/RequirementsServiceTest.cs

```diff
@@ -0,0 +1,111 @@
+﻿using BussinesObject.Api.RestEntities.RequirementModels;
+using BussinesObject.Api.Services;
+using BussinesObject.Api.Steps;
+using BussinesObject.Api.Utils;
+using Core.Common;
+
+namespace Test.ApiTests;
+
+[TestFixture]
+public class RequirementsServiceTest : BaseTest
+{
+    private static readonly RequirementsService service = RequirementsRequest.Service;
+
+    [Test]
+    public void GetAllRequirements()
+    {
+        var respon = service.GetAllRequirements();
+        RequestHelper.CheckResponse(respon, false, System.Net.HttpStatusCode.OK);
+    }
+
+    [Test]
+    public void GetTags()
+    {
+        var respon = service.GetTagsByRequirement(1);
+        RequestHelper.CheckResponse(respon, false, System.Net.HttpStatusCode.OK);
+    }
+
+    [Test]
+    public void GetTestCases()
+    {
+        var respon = service.GetTestCasesByRequirement(1);
+        RequestHelper.CheckResponse(respon, false, System.Net.HttpStatusCode.OK);
+    }
+
+    [Test]
+    public void GetRequirement()
+    {
+        var respon = service.GetRequirement(1);
+        RequestHelper.CheckResponse(respon, false, System.Net.HttpStatusCode.OK);
+    }
+
+    [Test]
+    public void GetRequirementTypes()
+    {
+        var respon = service.GetRequirementTypes();
+        RequestHelper.CheckResponse(respon, false, System.Net.HttpStatusCode.OK);
+    }
+
+    [Test]
+    public void GetAllTags()
+    {
+        var respon = service.GetAllTags();
+        RequestHelper.CheckResponse(respon, false, System.Net.HttpStatusCode.OK);
+    }
+
+    [Test]
+    public void CreateAndDeleteRequirement()
+    {
+        var name = DataGenerator.GetSentence();
+        var description = DataGenerator.GetText();
+        var projectId = Core.Settings.Settings.API.MainProjectId;
+
+        var model = new CreateRequirementModel()
+        {
+            ProjectId = projectId,
+            Name = name,
+            Description = description,
+            TypeId = 3,
+            Tags = new() { "test" }
+        };
+
+        var responModel = RequirementsRequest.CreateRequirement(model);
+
+        var respon = service.DeleteRequirement(responModel!.Id);
+        RequestHelper.CheckResponse(respon, true, System.Net.HttpStatusCode.NoContent);
+    }
+
+    [Test]
+    public void UpdateRequirement()
+    {
+        var name = DataGenerator.GetSentence();
+        var description = DataGenerator.GetText();
+        var requirementId = 7;
+
+        var model = new CreateRequirementModel()
+        {
+            Name = name,
+            Description = description,
+            TypeId = 4,
+        };
```

Comment:
use the builder for it

Reply:
This model is used in several tests. Different fields are filled in there depending on the test, I think it’s not very good to make a generator constructor for each test, it’s easier to fill in the necessary fields right away.